plugins/jenkins-backend/src/service/jenkinsApi.ts

```diff
@@ -75,13 +75,16 @@ export class JenkinsApiImpl {
     const projects: BackstageProject[] = [];
 
     if (branch) {
-      // we have been asked to filter to a single branch.
       // Assume jenkinsInfo.jobFullName is a folder which contains one job per branch.
       // TODO: extract a strategy interface for this
-      const job = await client.job.get({
-        name: `${jenkinsInfo.jobFullName}/${branch}`,
-        tree: JenkinsApiImpl.jobTreeSpec.replace(/\s/g, ''),
-      });
+      const job = await Promise.any(
+        branch.split(/,/g).map(name =>
```

Comment:
Do you feel strongly about it still accepting `string`s? The code is a beat cleaner when the options are either `string[]` or `undefined` instead of a third option.

Reply:
Wouldn't this be a breaking change without still supporting `string | string[]`? Best to keep those to a minimum. We could add a log message saying that `string` has been deprecated when a `string` type is passed?